src/bin/ion/commands/dump.rs

```diff
@@ -108,7 +105,7 @@ fn write_all_values<W: Writer>(reader: &mut Reader, writer: &mut W) -> IonResult
     let mut annotations = vec![];
     loop {
         match reader.next()? {
-            StreamItem::Value(ion_type) => {
+            StreamItem::Value(ion_type) | StreamItem::Null(ion_type) => {
```

Comment:
🗺️ This fixes a bug in `null` handling when the `null` has annotations or a field ID.